### PR TITLE
Compression doc structure refactor

### PR DESCRIPTION
**Pull request by suiguoxin**
- Reuse class doc string in "Read the Docs" as in NAS
- remove redundant parameter 'optimizer ' from Pruners ('level', 'slim', 'fpgm', 'l1', 'l2') and change related unit test accordingly;
- fix some typos

### CODE REVIEW

File: docs/en_US/Compressor/Pruner.md [hunk 1]
```diff
@@ -283,23 +309,27 @@ pruner = TaylorFOWeightFilterPruner(model, config_list, statistics_batch_num=1)
 pruner.compress()
 ```
 
-You can view example for more information
-
-### User configuration for TaylorFOWeightFilterPruner
 
-- **sparsity:** How much percentage of convolutional filters are to be pruned.
-- **op_types:** Currently only Conv2d is supported in TaylorFOWeightFilterPruner.
+#### User configuration for TaylorFOWeightFilterPruner
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Pruner.md:313]: TaylorFOWeightFilterPruner -> TaylorFOWeightFilter Pruner

File: docs/en_US/Compressor/Pruner.md [hunk 2]
```diff
@@ -223,16 +246,18 @@ pruner.compress()
 
 Note: ActivationAPoZRankFilterPruner is used to prune convolutional layers within deep neural networks, therefore the `op_types` field supports only convolutional layers.
 
-You can view example for more information
+You can view [example](https://github.com/microsoft/nni/blob/master/examples/model_compress/model_prune_torch.py) for more information.
 
 ### User configuration for ActivationAPoZRankFilterPruner
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Pruner.md:251]: ActivationAPoZRankFilterPruner -> ActivationAPoZRankFilter Pruner

File: docs/en_US/Compressor/Pruner.md [hunk 3]
```diff
@@ -342,14 +372,21 @@ PyTorch code
...
-* **initial_sparsity:** This is to specify the sparsity when compressor starts to compress
-* **final_sparsity:** This is to specify the sparsity when compressor finishes to compress
-* **start_epoch:** This is to specify the epoch number when compressor starts to compress, default start from epoch 0
-* **end_epoch:** This is to specify the epoch number when compressor finishes to compress
-* **frequency:** This is to specify every *frequency* number epochs compressor compress once, default frequency=1
+
+##### PyTorch
+
+```eval_rst
+..  autoclass:: nni.compression.torch.AGP_Pruner
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Pruner.md:382]: could you help change all `AGP_Pruner` to `AGPPruner`? this pruner's name is different from all others...

File: docs/en_US/Compressor/Pruner.md [hunk 4]
```diff
@@ -674,8 +575,11 @@ The above configuration means that there are 5 times of iterative pruning. As th
 
 #### User configuration for LotteryTicketPruner
 
-* **prune_iterations:** The number of rounds for the iterative pruning, i.e., the number of iterative pruning.
-* **sparsity:** The final sparsity when the compression is done.
+##### PyTorch
+
+```eval_rst
+..  autoclass:: nni.compression.torch.ADMMPruner
```
Comment (QuanluZhang) [inline docs/en_US/Compressor/Pruner.md:581]: this is not ADMMPruner

File: src/sdk/pynni/nni/compression/torch/pruning/agp.py [hunk 5]
```diff
@@ -14,28 +22,24 @@
...
+        Supported keys:
+            - initial_sparsity: This is to specify the sparsity when compressor starts to compress.
+            - final_sparsity: This is to specify the sparsity when compressor finishes to compress.
+            - start_epoch: This is to specify the epoch number when compressor starts to compress, default start from epoch 0.
+            - end_epoch: This is to specify the epoch number when compressor finishes to compress.
+            - frequency: This is to specify every *frequency* number epochs compressor compress once, default frequency=1.
+    optimizer: torch.optim.Optimizer
+        Optimizer used to train model.
+    pruning_algorithm: str
+        Algorithms being used to prune model.
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/pruning/agp.py:39]: better to list all the options here

File: src/sdk/pynni/nni/compression/torch/pruning/one_shot.py [hunk 6]
```diff
@@ -85,12 +85,32 @@ def calc_mask(self, wrapper, wrapper_idx=None):
...
+    """
+    Parameters
+    ----------
+    model : torch.nn.module
+        Model to be pruned
+    config_list : list
+        Supported keys:
+            - sparsity : This is to specify the sparsity operations to be compressed to.
+            - op_types : Only BatchNorm2d is supported in Slim Pruner.
+    """
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/pruning/one_shot.py:111]: this docstring is put above `__init__`, @chicm-ms @liuzhe-lz do you think this is a good practice?

File: docs/en_US/Compressor/Pruner.md (hunk 1 above)
Comment (suiguoxin) [reply docs/en_US/Compressor/Pruner.md]: fixed

File: docs/en_US/Compressor/Pruner.md (hunk 2 above)
Comment (suiguoxin) [reply docs/en_US/Compressor/Pruner.md]: fixed

File: docs/en_US/Compressor/Pruner.md (hunk 3 above)
Comment (suiguoxin) [reply docs/en_US/Compressor/Pruner.md]: fixed

File: docs/en_US/Compressor/Pruner.md (hunk 4 above)
Comment (suiguoxin) [reply docs/en_US/Compressor/Pruner.md]: fixed

File: src/sdk/pynni/nni/compression/torch/pruning/agp.py (hunk 5 above)
Comment (suiguoxin) [reply src/sdk/pynni/nni/compression/torch/pruning/agp.py]: fixed

File: src/sdk/pynni/nni/compression/torch/pruning/one_shot.py (hunk 6 above)
Comment (suiguoxin) [reply src/sdk/pynni/nni/compression/torch/pruning/one_shot.py:111]: This is to keep consistent with NAS classes and generate correct docstring in readthedocs.

Comment (chicm-ms) [reply src/sdk/pynni/nni/compression/torch/pruning/one_shot.py:111]: I think this is a good practice since `__init__` is the constructor of the class, I checked pytorch and tensorflow, they put docstring of class before `__init__`,  including the parameters of `__init__`.   I was not aware of this before.  For example: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/base_layer.py#L114

Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/pruning/one_shot.py:111]: got it, thanks.

File: src/sdk/pynni/nni/compression/torch/pruning/one_shot.py [hunk 7]
```diff
@@ -85,12 +85,32 @@ def calc_mask(self, wrapper, wrapper_idx=None):
             return None
 
 class LevelPruner(OneshotPruner):
-    def __init__(self, model, config_list, optimizer=None):
-        super().__init__(model, config_list, pruning_algorithm='level', optimizer=optimizer)
+    """
+    Parameters
+    ----------
+    model : torch.nn.module
```
Comment (chicm-ms) [inline src/sdk/pynni/nni/compression/torch/pruning/one_shot.py:91]: torch.nn.module  should be `torch.nn.Module`

File: src/sdk/pynni/nni/compression/torch/pruning/simulated_annealing_pruner.py [hunk 8]
```diff
@@ -22,62 +22,56 @@
...
+                        correct += pred.eq(target.view_as(pred)).sum().item()
+                accuracy = correct / len(val_loader.dataset)
+                return accuracy
+    optimize_mode : str
+        Optimize mode, `maximize` or `minimize`, by default `maximize`.
+    base_algo : str
+        Base pruning algorithm. `level`, `l1` or `l2`, by default `l1`. Given the sparsity distribution among the ops,
+        the assigned `base_algo` is used to decide which filters/channels/weights to prune.
+    start_temperature : float
+        Simualated Annealing related parameter.
```
Comment (chicm-ms) [inline src/sdk/pynni/nni/compression/torch/pruning/simulated_annealing_pruner.py:60]: suggest to provide more description for `start_temperature`, `stop_temperature`, and `cool_down_rate`

File: src/sdk/pynni/nni/compression/torch/pruning/one_shot.py [hunk 9]
```diff
@@ -85,12 +85,32 @@ def calc_mask(self, wrapper, wrapper_idx=None):
...
+    Parameters
+    ----------
+    model : torch.nn.Module
+        Model to be pruned
+    config_list : list
+        Supported keys:
+            - sparsity : This is to specify the sparsity operations to be compressed to.
+            - op_types : Operation types to prune.
+    """
+    def __init__(self, model, config_list):
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/compression/torch/pruning/one_shot.py:98]: `optimizer=None` should not be removed. The example does not work without this parameter.
